letsencrypt-auto-source/tests/centos6_tests.sh

```diff
@@ -4,75 +4,90 @@ yum update -y > /dev/null
 yum install -y centos-release-scl > /dev/null
 yum install -y python27 > /dev/null 2> /dev/null
 
+LE_AUTO_PRE_0_37="certbot/letsencrypt-auto-source/letsencrypt-auto_pre_0_37"
 LE_AUTO="certbot/letsencrypt-auto-source/letsencrypt-auto"
 
 # we're going to modify env variables, so do this in a subshell
 (
 source /opt/rh/python27/enable
 
 # ensure python 3 isn't installed
-python3 --version 2> /dev/null
+python3 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -eq 0 ]; then
-  error "Python3 is already installed."
+  echo "ERROR: Python3 is already installed."
   exit 1
 fi
 
 # ensure python2.7 is available
-python2.7 --version 2> /dev/null
+python2.7 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -ne 0 ]; then
-  error "Python3 is not available."
+  echo "ERROR: Python2.7 is not available."
   exit 1
 fi
 
 # bootstrap, but don't install python 3.
 "$LE_AUTO" --no-self-upgrade -n > /dev/null 2> /dev/null
 
 # ensure python 3 isn't installed
-python3 --version 2> /dev/null
+python3 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -eq 0 ]; then
-  error "letsencrypt-auto installed Python3 even though Python2.7 is present."
+  echo "ERROR: letsencrypt-auto installed Python3 even though Python2.7 is present."
   exit 1
 fi
 
-echo ""
 echo "PASSED: Did not upgrade to Python3 when Python2.7 is present."
 )
 
 # ensure python2.7 isn't available
-python2.7 --version 2> /dev/null
+python2.7 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -eq 0 ]; then
-  error "Python2.7 is still available."
+  error "ERROR: Python2.7 is still available."
   exit 1
 fi
 
 # Skip self upgrade due to Python 3 not being available.
 if ! "$LE_AUTO" 2>&1 | grep -q "WARNING: couldn't find Python"; then
-  echo "Python upgrade failure warning not printed!"
+  echo "ERROR: Python upgrade failure warning not printed!"
   exit 1
 fi
 
```

Comment:
What do you think about adding a test here to test the case when certbot-auto is run non-interactively and therefore choses not to run on Python 3.6. To relevant code for this is at https://github.com/certbot/certbot/blob/ada2f5c767f11b60e246fa1a5130fa67d64f6a78/letsencrypt-auto-source/letsencrypt-auto#L1080.

To do this, I think we can:

1. Rename the script to something other than `letsencrypt-auto` which implies `--non-interactive`.
2. Run something like `"$CB_AUTO | tee /dev/null"` which should make `test -t 1` false.

To test, we could check that the version output is still the old version of certbot-auto and that SCL is not installed.

Reply:
I implemented this test.